crates/atuin/src/command/client/external.rs

```diff
@@ -0,0 +1,72 @@
+use std::fmt::Write as _;
+use std::{ffi::OsString, process::Command};
+use std::{io, process};
+
+use clap::CommandFactory;
+use clap::builder::{StyledStr, Styles};
+use eyre::Result;
+
+use crate::Atuin;
+
+pub fn run(args: &[OsString]) -> Result<()> {
+    let subcommand = args[0].to_str().unwrap();
+    let bin = format!("atuin-{subcommand}");
+    let mut cmd = Command::new(&bin);
+    cmd.args(&args[1..]);
+
+    let spawn_result = match cmd.spawn() {
+        Ok(child) => Ok(child),
+        Err(e) => match e.kind() {
+            io::ErrorKind::NotFound => {
+                let output = render_not_found(subcommand, &bin);
+                Err(output)
+            }
+            _ => Err(e.to_string().into()),
+        },
+    };
+
+    match spawn_result {
+        Ok(mut child) => {
+            let status = child.wait()?;
+            if status.success() {
+                Ok(())
+            } else {
+                process::exit(status.code().unwrap_or(1));
+            }
+        }
+        Err(e) => {
+            println!("{}", e.ansi());
+            process::exit(1);
+        }
+    }
+}
+
+fn render_not_found(subcommand: &str, bin: &str) -> StyledStr {
+    let mut output = StyledStr::new();
+    let styles = Styles::styled();
```

Comment:
@ellie Do we need to handle disabling colors if the user has them disabled, or does Clap handle that for us?

Reply:
I believe clap should handles that part for us - should be easy to test